src/middlewares/facilityMiddleware.js

```diff
@@ -30,4 +32,26 @@ export default class FacilityMiddleware {
       }
     };
   }
+
+  /**
+    * Middleware method to authenticate facility.
+    * @param {object} req - The request from the endpoint.
+    * @param {object} res - The response returned by the method.
+    * @param {object} next - Call the next operation.
+    *@returns {object} - Returns an object (error or response).
+    * @memberof FacilityMiddleware
+    *
+    */
+  static async verifyFacility(req, res, next) {
+    try {
+      let { facilityId } = req.params;
+      facilityId = Number(facilityId);
+      const { dataValues: facility } = await findFacilityById(facilityId);
+      if (facility) {
+        return next();
+      }
+    } catch (err) {
+      errorResponse(res, { code: 401, message: err.message });
```

Comment:
Why 401 here? I assume that if the facility is not found, it should be a 404

Reply:
Also, can anyone update a facility regardless of whether he/she created the facility?

Reply:
No only a superAdmin can update it. The update frontend works with click boxes, so if a new facility is clicked, it will be part of the new update (if a facility is un-clicked, it will be dropped in the new update). original amenities are ticked by default.